etc/analytics.api.md

```diff
@@ -26,6 +26,9 @@ export interface AnalyticsEventService {
     with(payload: EventPayload): AnalyticsEventService;
 }
 
+// @public
+export function analyticsGTM(): Promise<string>;

```

Comment:
Why do we need to create another function just for GTM?

Reply:
Originally we tried to just use the original function, however google tag manager uses Sandboxed Javascript which has limited functionality. We can inject the script, but then we cannot call with because it returns a type not accepted by GTM, which results in undefined. After this we considered sending a post request directly to the events api instead, but that is not supported either (only get requests are). 

The workaround is for google tag manager to store the event payload in the window object as analyticsEventPayload. It will have two children, config and payload. Then GTM will call this function which reads from the window object and calls report.